src/main/java/core/basesyntax/MyLinkedList.java

```
@@ -1,47 +1,160 @@
+        Node<T> predecesor = succesor.prev;
+        Node<T> newNode = new Node<>(predecesor, value, succesor);
+        succesor.prev = newNode;
+        if (predecesor == null) {
+            first = newNode;
+        } else {
+            predecesor.next = newNode;
+        }
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
+        if (list == null) {
+            throw new NullPointerException("The provided list is null");
+        }
+        for (T item : list) {
+            add(item);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkElementIndex(index);
+        return findNodeByIndex(index).item;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        checkElementIndex(index);
+        Node<T> node= findNodeByIndex(index);
+        T oldValue = node.item;
+        node.item = value;
+        return oldValue;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkElementIndex(index);
+        return unlink(findNodeByIndex(index));
     }
 
     @Override
     public boolean remove(T object) {
+        Node<T> current = first;
+        while (current != null) {
+            if (object == null ? current.item == null : object.equals(current.item)) {
+                unlink(current);
+            }
+            current = current.next;
+        }
         return false;
     }
 
     @Override
     public int size() {
-        return 0;
+        return size;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        return size == 0;
+    }
+
+    private void checkElementIndex(int index) {
+        if (index < 0 || index >= size) {
+            throw new IndexOutOfBoundsException("Index: " + index + ", Size: " + size);
+        }
+    }
+
+    private void checkPositionIndex(int index) {
+        if (index < 0 || index <size) {
```

The condition `index < size` is incorrect for checking a position index. An element can be added at any index from `0` up to `size` (inclusive). The check should throw an exception if `index < 0` or `index > size`.